structured_prediction_baselines/modules/structured_score/sequence_tagging/self_attention.py

```diff
@@ -45,3 +66,31 @@ def forward(
 
         # reduction = "max" (Default)
         return attention_output.amax(dim=3).sum(2)
+
+    def _get_attention_mask(self, n_samples, mask: torch.Tensor):
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        batch_size, seq_length = mask.shape
+        attention_mask = torch.BoolTensor(batch_size * n_samples, seq_length, seq_length).fill_(False)
+        attention_mask = attention_mask.to(device=device)
+        masked_length = mask.sum(dim=-1)
```

Comment:
If the masked_length simply gives you the length of the contiguous 1s in the mask, e.g. 3 for the following mask [1110000000], then I think this code is correct.

Reply:
right, a line of comment to explain the shape and values of mask will be helpful.